### PR TITLE
Fix critical npm advisories by force updating transient dependencies immer and set-value

### PR DESCRIPTION
**Pull request by petemill**
- Use 'npm ci' and not 'npm install' so that we can replicate package-lock.json exactly and include forced transient dependencies

- `npx npm-force-resolutions` was used to apply package.json `resolutions` field to package-lock.json. `npm ci` should be used to replicate package-lock.json. `npm install` will just overwrite with insecure versions of transient dependencies.

Whenever `npm install` is used to add a dependency, then `npx npm-force-resolutions` will need to be re-run in order to avoid pushing a version of package-lock.json that removes these. PR reviews should catch this. Alternatively we can add a pre-install hook to run `npx npm-force-resolutions`. I'd like to keep it simple for now.

**DO NOT MERGE before https://github.com/brave/web-discovery-project/pull/70 is accepted and `brave/DEPS` is updated to include it.**

Resolves https://github.com/brave/brave-browser/issues/19299

## Submitter Checklist:

- [ ] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [ ] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [ ] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

File: DEPS
Comment (petemill) [inline DEPS:17]: this will be changed back once (or if) https://github.com/brave/web-discovery-project/pull/70 is accepted

**Comment by diracdeltas** [comment]
i'm in favor of adding `npm-force-resolutions` as a dev dependency and then adding it as a preinstall script like the example in https://www.npmjs.com/package/npm-force-resolutions - otherwise this will break again the next time someone makes a change to dependencies, and it might not be caught in CI because audit_deps is not run there unless they add the `run-audit-deps` label. otherwise this lgtm pending the WDP merge.

**Comment by petemill** [comment]
@diracdeltas the preinstall script does seem to solve the issue in that the node_modules dir will always be installed with the fixed transitive dependencies, whether using `npm ci` or `npm install`. Annoyingly if a dev runs `npm install` to add a dependency then they'll still end up pushing a version of package-lock.json that restores the non-fixed transient dependencies (even though they'll never get installed). Might be annoying for some devs. Maybe we need to also run it as a postinstall step. Or keep the original package-lock.json and don't use `npm ci`. I think I'll do that - less complicated.

`npm ci` and `npm install` with that preinstall script leave the package-lock.json in different states. The former leaves it with the "fixed" deps and the latter with the "original" deps. Both leave node_modules in the state that does not have the vulnerabilities, so I'll remove npm ci from this PR. We can address possible transient pinning as a follow-up.

**Comment by diracdeltas** [comment]
not sure if you are running into this particular issue, but there is also this workaround: https://stackoverflow.com/a/67446313

**Comment by petemill** [comment]
@diracdeltas haven't run in to that issue, but will keep it in mind. Seems to be working pretty well now with the latest and using both regular `npm install` and `npm install --save-dev [new-dep]` doesn't interfere with npm-force-resolution or bring back unwanted transient deps.

**Review by iefremov — APPROVED** [review]
unblocking the DEPS review

**Review by bsclifton — APPROVED** [review]
++

**Review by diracdeltas — APPROVED** [review]
lgtm

**Comment by petemill** [comment]
Pushed update to WDP and to set-value dep, waiting for CI. Once green, can merge.

**Comment by petemill** [comment]
CI is all passed, issues were only unrelated artifact upload

**Pull request merged**
